lib/board.rb

```diff
@@ -0,0 +1,63 @@
+class Board
+
+  attr_accessor :cells
+
+  @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+
+  def initialize
+    reset
+  end
+
+  def reset
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+      puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+      puts "-----------"
+      puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+      puts "-----------"
+      puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(pos)
+    @cells[pos.to_i - 1]
+  end
+
+  def full?
+    @cells.all? do |token|
+      token == "X" || token == "O"
+    end
+  end
+
+  def turn_count
+    counter = 0
+    @cells.each do |cell|
+      if cell == "X" || cell == "O"
+        counter += 1
+      end
+    end
+    counter
+  end
+
+  def taken?(move)
+    if @cells[move.to_i-1] == "X" || @cells[move.to_i-1] == "O"
+      true
+    elsif @cells[move.to_i-1] == " " || @cells[move.to_i-1] == ""
+      false
+    end
```

Comment:
Is there any other possible value for `@cells[move.to_i-1]` that would belong in an `else` clause? If so, would the return value (`false`) change?
I propose that you could replace your `elsif` with `else` and this method would be logically equivalent.
```
def taken?(move)
  if @cells[move.to_i-1] == "X" || @cells[move.to_i-1] == "O"
    true
  else
    false
  end
end
```
Now, generally you want to avoid a dichotomous method that returns boolean literals based on a single condition. We could utilize the implicit return value of the boolean expression and simplify the updated method to:
```
def taken?(move)
  @cells[move.to_i-1] == "X" || @cells[move.to_i-1] == "O"
end
```